osu.Game/Overlays/ChatOverlay.cs

```diff
@@ -478,11 +478,35 @@ private void postMessage(TextBox textbox, bool newText)
                 return;
             }
 
+            bool isAction = false;
+
             if (postText[0] == '/')
             {
-                // TODO: handle commands
-                target.AddNewMessages(new ErrorMessage("Chat commands are not supported yet!"));
-                return;
+                string unhandeledParameters = postText.Substring(1);

```

Comment:
I think this feels cleaner personally:

![image](https://user-images.githubusercontent.com/191335/30747391-11189958-9fe8-11e7-9b3c-bb9997b599f9.png)

Reply:
Don't you think that there should be a function that returns the next parameter? I image this is more useful for future commands.

Reply:
i think each command should probably be responsible for its own parsing, and in most cases it's going to be more natural to get an `args[]` and take action based on the length of the array.

Imagine you have a command which could be executed as
`/invite xxx [yyy]`
where `[yyy]` is optional.

using your command you'd have to do something like

```
var arg1 = cutNextparameter(ref args);
var arg2 = cutNextParameter(ref args);

if (string.IsNullOfEmpty(arg2)
    //pattern 1
else
    //pattern 2
```

instead of

````
var args = content.Split(' ');

switch (args.Length)
{
    case 1:
        //pattern 1
       break;
    case 2:
        //pattern 2
       break;
}
````

i dunno